src/bom/bitcode.rs

```diff
@@ -340,6 +340,10 @@ fn build_bitcode_arguments(chan : &mut mpsc::Sender<Option<Event>>,
     }
     bcgen_op.push_arg("-Wno-error=unused-command-line-argument");
 
+    // C99 and later do not support implicit function declarations, but be more
+    // permissive (as GCC seems to be).
+    bcgen_op.push_arg("-Wno-error=implicit-function-declaration");
```

Comment:
I don't have any inherent problem with this, but I do have a question: would it be more robust(?) to try to infer the language version being targeted by the "real" compiler and just use that?

For example, maybe we could tell that the target is c89 or gnu89 and use `-std=c89` instead? I don't know the right answer to how, but I could imagine a world where we create a little C file that we preprocess and run (or just look at preprocessor definitions) to see the target language version.

That does seem hard, but it seems at least worth a discussion...

Reply:
My main concern would be getting that wrong.  There might not be an explicit `-std` specification on the command-line, and so I suppose the default would be based on the compiler version.  We could trace mainstream `gcc` versions, but modified `gcc` or other compilers (`icc`, `msvc`) would be difficult to track.

It also doesn't seem to be solely a C std issue. For example:
```
$ gcc --version
gcc (GCC) 13.2.0
$ clang --version
clang version 16.0.6
$ echo 'int main() { printf("hello"); }' | gcc -std=c99 -xc -
# generates warning, but completes successfully
$ echo 'int main() { printf("hello"); }' | clang -std=c99 -xc -
# generates error and exits with non-zero error
```

I think if we take the policy that the behind-the-scenes `clang` invocation should *always* be more permissive than the primary compiler that we should be OK: it's up to the primary compiler to reject the user's code based on the build configuration set by the user, and the build-bom-generated clang should just get the bitcode for us without worrying about code validation.